src/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -130,11 +131,68 @@ QString DBusService::getLogs() {
 }
 
 bool DBusService::run(Op op, const Config& config) {
-  return WgQuickProcess::run(
-      op, config.m_privateKey, config.m_deviceIpv4Address,
-      config.m_deviceIpv6Address, config.m_serverIpv4Gateway,
-      config.m_serverIpv6Gateway, config.m_serverPublicKey,
-      config.m_serverIpv4AddrIn, config.m_serverIpv6AddrIn,
-      config.m_allowedIPAddressRanges.join(", "), config.m_serverPort,
-      config.m_ipv6Enabled);
+
+  // Make the config file
+
+  QTemporaryDir tmpDir;
+
+  QString confFile = WgQuickProcess::writeWgConfigFile(
+    tmpDir,
+    config.m_privateKey, config.m_deviceIpv4Address,
+    config.m_deviceIpv6Address, config.m_serverIpv4Gateway,
+    config.m_serverIpv6Gateway, config.m_serverPublicKey,
+    config.m_serverIpv4AddrIn, config.m_serverIpv6AddrIn,
+    config.m_allowedIPAddressRanges.join(", "), config.m_serverPort,
+    config.m_ipv6Enabled);
+
+  if (confFile == WgQuickProcess::FAIL) {
+    return false;
+  }
+
+  // Make our copy of the bash script
+
+  // Read in bash file and write to tmp so can execute it.
+  // TODO: This seems cumbersome, but I didn't know a better way to do it
```

Comment:
Are you doing this because the script is included as a resource? What about if we install it as it is?

Here is an idea:
- in src.pro, only for Linux, we can add:
```
helperFile.path = /usr/share/mozillavpn
helperFile.files = ../linux/extra/helper.sh
INSTALLS += helperFile
```
- WgQuickProcess uses `/usr/share/mozillavpn/helper.sh` instead of `wg-quick`.